mars/oscar/backends/communication/core.py

```diff
@@ -43,25 +42,28 @@ def _check_scheme(scheme: str, types: Dict):
     return scheme
 
 
-def get_client_type(address: str) -> Type[Client]:
+def get_scheme(address: str) -> str:
     if '://' not in address:
         scheme = None
     else:
         scheme = urlparse(address).scheme
-    scheme = _check_scheme(scheme, _scheme_to_client_types)
+    return scheme
+
+
+def get_client_type(address: str) -> Type[Client]:
+    scheme = _check_scheme(get_scheme(address), _scheme_to_client_types)
     return _scheme_to_client_types[scheme]
 
 
 def get_server_type(address: str) -> Type[Server]:
-    if '://' not in address:
-        scheme = None
-    else:
-        scheme = urlparse(address).scheme
-    scheme = _check_scheme(scheme, _scheme_to_server_types)
+    scheme = _check_scheme(get_scheme(address), _scheme_to_server_types)
     return _scheme_to_server_types[scheme]
 
 
-def gen_internal_address(process_index: int) -> str:
+def gen_internal_address(process_index: int,
```

Comment:
`get_internal_address` looks a bit weird to be here, it should be backend related.

Reply:
Make it as a abstract method of `MainActorPool`